bot/cogs/filtering.py

```diff
@@ -126,6 +136,54 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
+    @staticmethod
+    def get_name_matches(name: str) -> List[re.Match]:
+        """Check bad words from passed string (name). Return list of matches."""
+        matches = []
+        for pattern in WATCHLIST_PATTERNS:
+            match = pattern.search(name)
+            if match:
+                matches.append(match)
+        return matches
+
+    async def check_send_alert(self, member: Member) -> bool:
+        """When there is less than 3 days after last alert, return `False`, otherwise `True`."""
+        last_alert = await self.name_alerts.get(member.id)
+        if last_alert:
+            last_alert = datetime.utcfromtimestamp(last_alert)
+            if datetime.utcnow() - timedelta(days=DAYS_BETWEEN_ALERTS) < last_alert:
+                log.trace(f"Last alert was too recent for {member}'s nickname.")
+                return False
+
+        return True
+
+    async def check_bad_words_in_name(self, member: Member) -> None:
+        """Send a mod alert every 3 days if a username still matches a watchlist pattern."""
+        # Use lock to avoid race conditions
+        async with self.name_lock:
+            # Check does nickname have match in filters.
+            matches = self.get_name_matches(member.display_name)
+
+            if not matches or not await self.check_send_alert(member):
+                return
+
+            log.info(f"Sending bad nickname alert for '{member.display_name}' ({member.id}).")
+            log_string = (
+                f"**User:** {member.mention} (`{member.id}`)\n"
+                f"**Display Name:** {member.display_name}\n"
+                f"**Bad Matches:** {', '.join(match.group() for match in matches)}"
+            )
+            await self.mod_log.send_log_message(
+                icon_url=Icons.token_removed,
+                colour=Colours.soft_red,
+                title="Username filtering alert",
+                text=log_string,
+                channel_id=Channels.mod_alerts
+            )
```

Comment:
This could all use a little bit of spacing
```suggestion
            log.info(f"Sending bad nickname alert for '{member.display_name}' ({member.id}).")

            log_string = (
                f"**User:** {member.mention} (`{member.id}`)\n"
                f"**Display Name:** {member.display_name}\n"
                f"**Bad Matches:** {', '.join(match.group() for match in matches)}"
            )

            await self.mod_log.send_log_message(
                icon_url=Icons.token_removed,
                colour=Colours.soft_red,
                title="Username filtering alert",
                text=log_string,
                channel_id=Channels.mod_alerts
            )
```